pyccel/decorators.py

```diff
@@ -109,3 +110,37 @@ def allow_negative_index(f,*args):
     def identity(f):
         return f
     return identity
+
+class kernel:
+    """
+    Decorator for marking a Python function as a kernel.
+
+    This class serves as a decorator to mark a Python function
+    as a kernel function, typically used for GPU computations.
+    
+    Parameters
+    ----------
+    f : function
+        The function to which the decorator is applied.
+    """
+    def __init__(self, f):
+        self._f = f
+
+    def __getitem__(self, *args):
+        """
+        Get item method for the kernel class.
+
+        This method allows the kernel function to be invoked with
+        arguments using square bracket notation.
+
+        Parameters
+        ----------
+        *args : tuple
```

Comment:
Usually when annotating an unpacked argument the type specifies the type of one of the elements. Maybe you can just put `Any`.  I'm not sure there are any strong restrictions here

Reply:
This is described here:
https://peps.python.org/pep-0484/#arbitrary-argument-lists-and-default-argument-values

It is like this because documentation tells people how to call your function. It doesn't tell people the type of the object in your code.

So if you have
```python
def f(*args):
    pass
```
You would call it as:
```python
f(1,2,3.0)
```
There are no tuples here so tuples are irrelevant for the docstring

Reply:
fixed d6a4291